libs/librepcb/projecteditor/boardeditor/deviceinstancepropertiesdialog.cpp

```diff
@@ -135,6 +135,13 @@ void DeviceInstancePropertiesDialog::keyPressEvent(QKeyEvent* e)
     }
 }
 
+void DeviceInstancePropertiesDialog::accept()
+{
+    if (applyChanges()) {
```

Comment:
This is good so far. But in `buttonBoxClicked()` above, the `if (applyChanges())` can now be removed to avoid executing `applyChanges()` twice.